// vite-plugin-realtime-diff/index.js
"use strict";var s=Object.defineProperty;var g=Object.getOwnPropertyDescriptor;var v=Object.getOwnPropertyNames;var x=Object.prototype.hasOwnProperty;var w=(e,t)=>{for(var i in t)s(e,i,{get:t[i],enumerable:!0})},y=(e,t,i,o)=>{if(t&&typeof t=="object"||typeof t=="function")for(let r of v(t))!x.call(e,r)&&r!==i&&s(e,r,{get:()=>t[r],enumerable:!(o=g(t,r))||o.enumerable});return e};var b=e=>y(s({},"__esModule",{value:!0}),e);var W={};w(W,{realtimeDiff:()=>$});module.exports=b(W);var h=require("url");var n={"iphone-x":"375px","2k":"2048px",desktop:"1920px","1080p":"1920px","720p":"1280px","4k":"3840px","100%":"100%"};function m(e,t,i,o){return`
    <html>
            <head>
              <style>
              body {
                    margin: 0;
                    background: black;
              }
                .iframe-container {
                  position: relative;
                  width: 100%;
                  height: 15000px;
                  overflow-y: scroll;
                }
                iframe {
                  position: absolute;
                  top: 0;
                  left: 50%;
                  transform: translateX(-50%);
                  width: ${i};
                  height: 100%;
                  border: none;
                  pointer-events: none;
                  overflow: hidden;
                }
                .overlay {
                  mix-blend-mode: difference;
                }
                .width-form {
                  position: fixed;
                  top: 10px;
                  left: 10px;
                  z-index: 9999;
                }
              </style>
            </head>
            <body>
            <form action="/_diff" class="width-form">
                  <select name="width" onchange="this.form.submit()">
                    ${Object.keys(n).map(r=>`<option value="${r}" ${o===r?"selected":""}>${r}</option>`).join("")}
                  </select>
                </form>
                <div class="iframe-container">
                        <iframe id="frame1" src="${e}"></iframe>
                        <iframe id="frame2" src="${t}" class="overlay"></iframe>
                </div>
            </body>
          </html>
    `}function p(e){return n[e]||n["100%"]}function d(e){try{return new URL(e),!0}catch{return!1}}function $(e){return{name:"vite-plugin-realtime-diff",configureServer(t){t.middlewares.use((i,o,r)=>{if(!i.url){r();return}let{query:f}=(0,h.parse)(i.url,!0),a=`http://${i.headers.host}${i.url}`,c=i.url;if(e=new URL(e).origin+c,f._diff&&f._diff==="true"){a=a.replace("?_diff=true",""),e=e.replace("?_diff=true","");let l=f.width||"100%",u=p(l);if(!d(a)||!d(e)){o.statusCode=400,o.end("Please provide valid URLs in the plugin options.");return}o.setHeader("Content-Type","text/html"),o.end(m(a,e,u,l))}else r()})}}}0&&(module.exports={realtimeDiff});
